containers/src/Kokkos_DualView.hpp

```diff
@@ -1046,9 +1061,22 @@ class DualView : public ViewTraits<DataType, Properties...> {
       }
     };
 
-    constexpr bool has_execution_space = alloc_prop_input::has_execution_space;
+    if constexpr (alloc_prop_input::sequential_host_init) {
+      static_assert(alloc_prop_input::initialize,
+                    "DualView: SequentialHostInit isn't compatible with "
+                    "WithoutInitializing!");
+      static_assert(!alloc_prop_input::has_execution_space,
+                    "DualView: SequentialHostInit isn't compatible with "
+                    "providing an execution space instance!");
 
-    if constexpr (has_execution_space) {
+      if (sizeMismatch) {
+        sync<typename t_host::memory_space>();
+        ::Kokkos::resize(arg_prop, h_view, n0, n1, n2, n3, n4, n5, n6, n7);
+        d_view =
+            create_mirror_view_and_copy(typename t_dev::memory_space(), h_view);
```

Comment:
Should we be updating the modify flags for host and device here?

Reply:
`sync` should make sure that the flags are set appropriately and we then modify host and deviove view consistently.